hw1_oop/HW1_python2_OOP.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+
+
+class Chat:
+    default_start = datetime(1970, 1, 1)
+    
+    def __init__(self):
+        self._chat_history = []    # List of all messages in chat
+            
+    def show_last_message(self):
+        last_message = self._chat_history[0]
+        print(last_message.show())
+    
+    def get_history_from_time_period(self, start=None, end=None):
+        if start is None:
+            start = self.default_start
+        if end is None:
+            end = datetime.now()
+        self.start = start
+        self.end = end
+                
+        for message in self._chat_history:
+            if self.end >= message.datetime >= self.start:
+                print(message.show())           
+    
+    def show_chat(self, separator='\n'):
+        print(*[message.show() for message in self._chat_history], sep=separator)
+       
+    def _recieve(self, message):
+        if isinstance(message, Message):
+            message.datetime = datetime.now()    # Set attribute datetime when message is recieved by chat
+            self._chat_history.insert(0, message)
```

Comment:
В таких случаях лучше писать `else` в котором кидать исключение, так как юзер может получить None, передав не тот тип, и потом будет гадать, что пошло не так 

```suggestion
        if isinstance(message, Message):
            message.datetime = datetime.now()    # Set attribute datetime when message is recieved by chat
            self._chat_history.insert(0, message)
        else:
            raise TypeError(f"Method does not support type {type(message)}")
```